code/game/turfs/simulated/floor_icon.dm

```diff
@@ -77,11 +88,18 @@ var/global/image/no_ceiling_image = null
 	if(is_plating() && !(isnull(broken) && isnull(burnt))) //temp, todo
 		icon = 'icons/turf/flooring/plating.dmi'
 		icon_state = "dmg[rand(1,4)]"
-	else if(flooring)
-		if(!isnull(broken) && (flooring.flags & TURF_CAN_BREAK))
-			add_overlay(flooring.get_flooring_overlay("[flooring.icon_base]-broken-[broken]","broken[broken]"))
-		if(!isnull(burnt) && (flooring.flags & TURF_CAN_BURN))
-			add_overlay(flooring.get_flooring_overlay("[flooring.icon_base]-burned-[burnt]","burned[burnt]"))
+	else if(effective_flooring)
+		if(!isnull(broken) && (effective_flooring.flags & TURF_CAN_BREAK))
+			add_overlay(effective_flooring.get_flooring_overlay("[effective_flooring.icon_base]-broken-[broken]","broken[broken]"))
+		if(!isnull(burnt) && (effective_flooring.flags & TURF_CAN_BURN))
+			add_overlay(effective_flooring.get_flooring_overlay("[effective_flooring.icon_base]-burned-[burnt]","burned[burnt]"))
+
+	if (snow_layers)
+		if (snow_footprints.len)
+			for(var/F in snow_footprints)
+				add_overlay(image(icon = 'icons/turf/outdoors.dmi', icon_state = snow_footprints[F], dir = text2num(F)))
```

Comment:
`snow_footprints` is initialized to a list, if the list is empty this should just be a no-op, so I think you can drop the middle if statement (It'll look a touch cleaner and that's really about it)